external_services/lti.py

```diff
@@ -8,6 +8,7 @@
 from oauthlib.oauth1 import Client, SIGNATURE_HMAC, SIGNATURE_TYPE_BODY, \
     SIGNATURE_TYPE_QUERY
 import json
+from urllib.parse import urlsplit, urljoin
```

Comment:
Didn't notice before, but these should use the import ordering set in PEP 8: standard lib import, standard lib import from, 3rd party libs (import + from), project imports. Without fixing everything, you could add the new line in the start:

```python
from hashlib import md5
from urllib.parse import urlsplit, urljoin

from django.conf import settings
from django.core.exceptions import PermissionDenied
from django.utils.translation import get_language
from rest_framework.reverse import reverse
from rest_framework.settings import api_settings
from oauthlib.common import urldecode
from oauthlib.oauth1 import Client, SIGNATURE_HMAC, SIGNATURE_TYPE_BODY, \
    SIGNATURE_TYPE_QUERY
import json
```